data/datasource.go

```diff
@@ -426,6 +431,36 @@ func readConsul(source *Source, args ...string) ([]byte, error) {
 	return data, nil
 }
 
+func readK8S(source *Source, args ...string) ([]byte, error) {
+	if len(args) < 1 {
+		return nil, fmt.Errorf("a key is needed")
```

Comment:
Hmm... Why not allow a missing key, and return the whole `configmap.Data` object? Then you could do something like:

```
$ gomplate -d k=k8s:///default/map -i '{{ (ds "k").foo }}'
hello world
```

Where the configmap's data looks like:

```json
{ "foo": "hello world", "bar": "goodbye world" }
```

Obviously simply specifying `{{ds "k" "foo"}}` would have the same effect, but I think the flexibility is useful, especially when doing things like iterating though all of the keys in the map:

```
$ gomplate -d k=k8s:///default/map -i '{{ range $key, $value := (ds "k") }}{{ $key }} is {{ $value }}{{"\n"}}{{end}}'
bar is goodbye world
foo is hello world
```

Reply:
How do I have to return the whole configmap.Data? Shoud a I return de []byte json?

Reply:
Yeah, you'd just return it as a JSON-encoded `string` converted to a `[]byte`